pkg/vsphere/disk/disk_manager.go

```diff
@@ -421,3 +423,51 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 
 	return sysPath, nil
 }
+
+// AttachAndMount creates and attaches a vmdk as a non-persistent disk, mounts it, and returns the mount path.
+func (m *Manager) AttachAndMount(op trace.Operation, datastoreURI *object.DatastorePath) (string, error) {
+	config := NewNonPersistentDisk(datastoreURI)
+	d, err := m.CreateAndAttach(op, config)
```

Comment:
The latter is what we're after - I think you and I are in agreement. My question to @caglar10ur was whether create and attach is performing the ref counting - the name of the call is quite explicit about there being a create involved and looking at the code I don't see where we'd hit ref count increments.
In a related note this could block for some time if we hit the max count, and we should be checking `op` afterwards for cancellation @caglar10ur :
```go
	// ensure we abide by max attached disks limits
	m.maxAttached <- true
```